***
# <font color=green size=10>CURSO DE ESTADÍSTICA - PARTE 2</font>
***

***
### <font color=redlight size=8>Alumno: Pablo Matías Monsalvo G8-ONE</font>
***

# <font color=green>1 CONOCIENDO LOS DATOS</font>
***

## <font color=green>1.1 Dataset del proyecto</font>
***

### Encuesta Nacional por Muestreo de Hogares - 2015

La **Encuesta Nacional por Muestreo de Hogares - ENMH** en Brasil anualmente investiga, de manera permanente, características generales de la población, educación, trabajo, ingresos, vivienda y otras, con frecuencia variables, de acuerdo a las necesidades de información del país, tales como las características de migración, fecundidad, nupcialidad, salud, seguridad alimentaria, entre otros temas. La recopilación de estas estadísticas constituye, durante los 49 años de realización de la investigación, un importante instrumento para la formulación, validación y evaluación de políticas orientadas al desarrollo socioeconómico y la mejora de las condiciones de vida en Brasil.

### Fuente de los Datos

Instituto Brasileño de Geografía y Estadística (IBGE) Encuesta Nacional por Muestreo de Hogares(PNAD) en el 2015

### Variables utilizadas

> ### Ingresos
> ***

Ingresos mensuales del trabajo principal para personas de 10 años o más.

> ### Edad
> ***

Edad del residente en años en la fecha de referencia .

> ### Altura (elaboración propia)
> ***

Altura del habitante en metros.

> ### UF
> ***

Unidad Federal

|Código|Descripción|
|---|---|
|11|Rondônia|
|12|Acre|
|13|Amazonas|
|14|Roraima|
|15|Pará|
|16|Amapá|
|17|Tocantins|
|21|Maranhão|
|22|Piauí|
|23|Ceará|
|24|Rio Grande do Norte|
|25|Paraíba|
|26|Pernambuco|
|27|Alagoas|
|28|Sergipe|
|29|Bahia|
|31|Minas Gerais|
|32|Espírito Santo|
|33|Rio de Janeiro|
|35|São Paulo|
|41|Paraná|
|42|Santa Catarina|
|43|Rio Grande do Sul|
|50|Mato Grosso do Sul|
|51|Mato Grosso|
|52|Goiás|
|53|Distrito Federal|

> ### Sexo	
> ***

|Código|Descripción|
|---|---|
|0|Masculino|
|1|Femenino|

> ### Años de Estudio
> ***

|Código|Descripción|
|---|---|
|1|Sin educación o menos de 1 año|
|2|1 año|
|3|2 años|
|4|3 años|
|5|4 años|
|6|5 años|
|7|6 años|
|8|7 años|
|9|8 años|
|10|9 años|
|11|10 años|
|12|11 años|
|13|12 años|
|14|13 años|
|15|14 años|
|16|15 años o más|
|17|No determinado|
||No aplica|

> ### Color
> ***

|Código|Descripción|
|---|---|
|0|Indígena|
|2|Blanco|
|4|Negro|
|6|Amarillo|
|8|Pardo|
|9|Sin declarar|

#### <font color='red'>Nota</font>
***
> Se realizaron los siguientes tratamientos sobre los datos originales:
> 1. Se eliminaron los registros en los que **Ingreso** no era válido. (999 999 999 999);
> 2. Se eliminaron los registros en los que **Ingreso** eran missing;
> 3. Solo se consideraron los registros de **Personas de Referencia** de cada hogar (responsable del hogar).

## 3 - Preparando el ambiente

### Importando pandas y leyendo el dataset del proyecto

https://pandas.pydata.org/

In [1]:
import pandas, numpy, scipy, matplotlib

print('Versión de pandas -> %s' % pandas.__version__)
print('Versión de numpy -> %s' % numpy.__version__)
print('Versión de scipy -> %s' % scipy.__version__)
print('Versión de matplotlib -> %s' % matplotlib.__version__)

Versión de pandas -> 2.3.0
Versión de numpy -> 2.2.6
Versión de scipy -> 1.15.3
Versión de matplotlib -> 3.10.3


In [2]:
import pandas as pd

In [3]:
datos = pd.read_csv('datos.csv', encoding = 'latin-1')
datos.head()

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
0      11     0    23      8               12      800  1.603808
1      11     1    23      2               12     1150  1.739790
2      11     1    35      8               15      880  1.760444
3      11     0    46      2                6     3500  1.783158
4      11     1    47      8                9      150  1.690631

##  04 - Utilizando DataFrames

Con la biblioteca `pandas`, podemos leer conjuntos de datos en diferentes formatos y convertirlos en un `DataFrame`. Para un conjunto de datos en formato CSV, ¿cuál sería la función de pandas para leer estos datos?

### `.read_csv()`

¡Alternativa correcta! Recuerde anotar el tipo de separador de columna que utiliza el archivo CSV. Por defecto, la función read_csv() usa el parámetro `sep=','`. Si el separador es cualquier carácter que no sea la coma, necesitaremos informar esto para la función, por ejemplo: `read_csv('data.csv', sep =';')`.





---
---

##  05 - Distribución Binomial


# <font color=green>2 DISTRIBUCIONES DE PROBABILIDAD</font>
***

## <font color=red>Problema</font>
***

En una competencia para ocupar un puesto de científico de datos, tenemos un total de **10 preguntas** de opción múltiple con **3 alternativas posibles** en cada pregunta. **Cada pregunta tiene el mismo valor.** Suponga que un candidato decide aventurarse sin haber estudiado nada. Decide hacer la prueba con los ojos vendados y adivinar todas las respuestas. Suponiendo que la prueba **vale 10 puntos y la puntuación de corte es 5**, obtenga la probabilidad de que este candidato **acierte 5 preguntas** y también la probabilidad de que el candidato **pase a la siguiente etapa del proceso de selección**.

## <font color=green>2.1 Distribución Binomial</font>
***

Un evento **binomial** se caracteriza por la posibilidad de que solo ocurran dos categorías. Estas categorías juntas representan todo el espacio muestral, siendo también mutuamente excluyentes, es decir, la ocurrencia de una implica la no ocurrencia de la otra.

En el análisis estadístico, el uso más común de la distribución binomial es para resolver problemas que involucran situaciones de **éxito** y **fracaso**.

# $$P(k)=\binom{n}{k} p^k q^{n-k}$$

Donde:

$p$ = probabilidad de éxito

$q = (1 - p)$ = probabilidad de fracaso

$n$ = número de eventos estudiados

$k$ = número de eventos deseados que tienen éxito

### Experimento Binomial

1. Realización de $n$ pruebas idénticas.

2. Las pruebas son independientes.

3. Solo son posibles dos resultados, por ejemplo: Verdadero o falso; Cara o cruz; Éxito o fracaso.

4. La probabilidad de éxito está representada por $p$ y la probabilidad de fracaso por $1-p = q$. Estas probabilidades no cambian de una prueba para otra.

### Media de la distribución binomial

El valor esperado o la media de la distribución binomial es igual al número de experimentos realizados multiplicado por la probabilidad de que ocurra el evento.

# $$\mu = n \times p$$

### Desviación estándar de la distribución binomial

La desviación estándar es la raíz quadrada del producto entre el número de experimentos, la probabilidad de éxito y la probabilidad de fracaso.

# $$\sigma = \sqrt{n \times p \times q}$$

### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.comb.html

In [4]:
from scipy.special import comb

### Combinaciones

El número de combinaciones de $n$ objetos, tomando $k$ cada vez, es:

# $$C_{k}^{n} = \binom{n}{k} = \frac{n!}{k!(n - k)!}$$

Donde

## $$n! = n\times(n-1)\times(n-2)\times...\times(2)\times(1)$$
## $$k! = k\times(k-1)\times(k-2)\times...\times(2)\times(1)$$

Por definición

## $$0! = 1$$

## <font color='blue'>Ejemplo: Mega Sena</font>

En una rueda de lotería Mega Sena tenemos un total de **60 números** para elegir, donde la apuesta mínima es de **seis números**. Usted que es curioso(a) decide calcular la probabilidad de ganar la Mega Sena con solo **un juego**. Para eso necesitamos saber cuántas **combinaciones de seis números se pueden formar con los 60 números disponibles**.

### $$C_{6}^{60} = \binom{60}{6} = \frac{60!}{6!(60 - 6)!}$$

In [5]:
combinaciones =comb(60, 6)
combinaciones

np.float64(50063860.0)

In [6]:
probabilidad = 1 / combinaciones
probabilidad


np.float64(1.997448858318156e-08)

In [7]:
print(f'{probabilidad:0.15f}')

0.000000019974489


## 06 - Exposición de premios de Alura

Supongamos que acabamos de crear un juego de lotería llamado Exposición de Premios de Alura. En este nuevo juego, el que apuesta marca 20 números, de los 25 disponibles en el boleto, y puede ganar hasta 1 millón de pesos.

Determine el número de combinaciones posibles (espacio muestral) y la probabilidad de ganar el premio jugando solo un boleto (considere sólo quince lugares decimales).



In [8]:
from scipy.special import comb

combinaciones = comb(25, 20)
probabilidad = 1 / combinaciones
print(f'Combinaciones = {combinaciones:.0f} e Probabilidad = {probabilidad:0.15f}')    

Combinaciones = 53130 e Probabilidad = 0.000018821757952


## ¡Alternativa correcta! El resultado anterior se puede obtener con el siguiente código:

Recordando que las combinaciones las obtenemos con la siguiente fórmula:

## Exposición de Premios de Alura, por Claude Sonnet 4 IA

Para calcular las combinaciones posibles de seleccionar 20 números de 25 disponibles:

$$C_{25}^{20} = \binom{25}{20} = \frac{25!}{20!(25-20)!} = \frac{25!}{20! \cdot 5!}$$

---

## Ejercicio: Exposición de Premios de Alura

Supongamos que acabamos de crear un juego de lotería llamado **Exposición de Premios de Alura**. En este nuevo juego, el que apuesta marca 20 números, de los 25 disponibles en el boleto, y puede ganar hasta 1 millón de pesos.

### Fórmula de Combinaciones

Para calcular el número de combinaciones posibles, utilizamos la fórmula:

$$C_{n}^{k} = \binom{n}{k} = \frac{n!}{k!(n - k)!}$$

Donde:
- $n$ = número total de elementos disponibles (25 números)
- $k$ = número de elementos a seleccionar (20 números)
- $n!$ = factorial de n
- $k!$ = factorial de k

### Aplicación al problema

En nuestro caso:
- $n = 25$ (números disponibles en el boleto)
- $k = 20$ (números que marca el apostador)

$$C_{25}^{20} = \binom{25}{20} = \frac{25!}{20!(25 - 20)!} = \frac{25!}{20! \cdot 5!}$$

$$C_{25}^{20} = \frac{25 \times 24 \times 23 \times 22 \times 21}{5 \times 4 \times 3 \times 2 \times 1}$$

$$C_{25}^{20} = \frac{6,375,600}{120} = 53,130$$

Por lo tanto, existen **53,130 combinaciones posibles** diferentes para seleccionar 20 números de los 25 disponibles.


##  07- Calculando probabilidades


## <font color='blue'>Ejemplo: Concurso para científico de datos</font>

En una competencia para ocupar un puesto de científico de datos, tenemos un total de **10 preguntas** de opción múltiple con **3 alternativas posibles** en cada pregunta. **Cada pregunta tiene el mismo valor.** Suponga que un candidato decide aventurarse sin haber estudiado nada. Decide tomar la prueba con los ojos vendados y adivinar todas las respuestas. Suponiendo que la prueba **vale 10 puntos y la puntuación de corte es 5**, obtenga la probabilidad de que este candidato **acierte 5 preguntas** y también la probabilidad de que el candidato **pase a la siguiente etapa del proceso de selección**.

### ¿Cuál es la cantidad de pruebas ($n$)?

In [9]:
n = 10
n

10

### ¿Las pruebas son independientes?

Sí. La opción elegida en una pregunta no influye en la opción elegida en otra pregunta.

### ¿Sólo dos resultados son posibles en cada prueba?

Sí. El candidato tiene dos posibilidades, ACERTAR o FALLAR una pregunta.

### ¿Cuál es la probabilidad de éxito ($p$)?

In [10]:
numero_de_alternativas_por_pregunta = 3
p = 1 / numero_de_alternativas_por_pregunta
p

0.3333333333333333

### ¿Cuál es la probabilidad de fracaso ($q$)?

In [11]:
q = 1 - p
q

0.6666666666666667

### ¿Cuál es el número total de eventos para tener éxito ($k$)?

In [12]:
k = 5
k

5

### Solución 1

In [13]:
probabilidad = (comb(10,5) * (p ** k) * (q ** (n - k)))
print(f'Probabilidad de acertar 5 preguntas = {probabilidad:.8f}')

Probabilidad de acertar 5 preguntas = 0.13656455


### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binom.html

In [14]:
from scipy.stats import binom

### Solución 2

In [15]:
probabilidad =  binom.pmf(k, n, p)
print(f'{probabilidad:.8f}')

0.13656455


### Obtenga la probabilidad de que el candidato apruebe

### $$P(acertar \geq 5) = P(5) + P(6) + P(7) + P(8) + P(9) + P(10)$$

In [16]:
binom.pmf(5,n,p) + binom.pmf(6,n,p) + binom.pmf(7,n,p) + binom.pmf(8,n,p) + binom.pmf(9,n,p) + binom.pmf(10,n,p)

np.float64(0.21312808006909476)

In [17]:
binom.pmf([5,6,7,8,9,10], n, p).sum()

np.float64(0.21312808006909476)

In [18]:
1 - binom.cdf(4, n, p)

np.float64(0.21312808006909523)

In [19]:
binom.sf(4, n, p)

np.float64(0.21312808006909525)

##  08 - Distribución de probabilidad binomial

Las siguientes alternativas describen las características básicas de un experimento binomial.

Marque las que sean correctas.

### 1-Solo dos resultados son posibles.

* ¡Alternativa correcta! Ejemplo: verdadero o falso; cara o cruz; éxito o fracaso.

### 2-Realización de n pruebas idénticas.

* ¡Alternativa correcta! Todas las pruebas realizadas deben tener la misma configuración.


##  09 - Lanzamiento de monedas

Una moneda perfectamente equilibrada se lanza al aire cuatro veces. Usando la distribución binomial, obtenga la probabilidad de que la moneda caiga con la cruz hacia arriba dos veces.

In [20]:
from scipy.stats import binom

p = 1 / 2      # Probabilidad de salir CRUZ 'la moneda tiene 2 caras'
n = 4          # Total de lanzamientos
k = 2          # Total de aciertos (CRUZ hacia arriba)

binom.pmf(k, n, p)

np.float64(0.375)

¡Alternativa correcta! Perfecto, tenga en cuenta que la expresión "perfectamente equilibrado" indica que la moneda utilizada es honesta, es decir, tiene la misma probabilidad de obtener cara o cruz. El siguiente código es un ejemplo de una solución:

##  10 - Lanzamiento de dados

Un dado perfectamente equilibrado se lanza hacia arriba diez veces. Usando la distribución binomial, obtenga la probabilidad de que los dados caigan con el número cinco hacia arriba al menos tres veces.

In [21]:
from scipy.stats import binom

p = 1 / 6   # Probabilidad de salir el número CINCO 'el dado tiene 6 caras'
n = 10      # Total de lanzamientos

print(f'{binom.sf(2, n, p):.2%}') 

22.48%


## Explicación por Claude Sonnet 4 IA:


In [22]:
# Importamos la distribución binomial de scipy.stats
from scipy.stats import binom

# PARÁMETROS DEL PROBLEMA
p = 1 / 6   # Probabilidad de éxito en cada lanzamiento
            # Probabilidad de que salga el número CINCO (1 cara de 6 posibles)

n = 10      # Número total de lanzamientos del dado

# CÁLCULO Y RESULTADO
print(f'{binom.sf(2, n, p):.2%}')
# binom.sf(2, n, p) = Probabilidad de obtener MÁS DE 2 éxitos
# Significa: P(X > 2) donde X = número de veces que sale el 5
# Incluye: P(X=3) + P(X=4) + P(X=5) + P(X=6) + P(X=7) + P(X=8) + P(X=9) + P(X=10)
# :.2% = Formato de porcentaje con 2 decimales

# INTERPRETACIÓN:
# "¿Cuál es la probabilidad de que el número 5 salga más de 2 veces en 10 lanzamientos?"
# Respuesta: 22.48%

22.48%


### En resumen:

* sf = "Survival Function" (más de X)
* 2 = valor de referencia (más de 2 veces)
* n=10 = total de lanzamientos
* p=1/6 = probabilidad de éxito por lanzamiento
* Resultado: 22.48% de probabilidad de que el 5 salga más de 2 veces

* Gracias por la corrección. El 22.48% es efectivamente el resultado correcto del cálculo.

##  11 - Aplicación de la media


## <font color='blue'>Ejemplo: Yincana</font>

Una ciudad del interior realiza una yincana cada año para recaudar fondos para el hospital de la ciudad. En la última competencia se sabe que la **proporción de mujeres participantes fue del 60%**. **El número total de equipos, con 12 miembros, inscritos en la yincana de este año es 30**. Con la información anterior, responda: ¿Cuántos equipos deberían estar formados por **8 mujeres**?

### Solución

In [23]:
p = 0.6
p
n = 12
n
k = 8
k
from scipy.stats import binom
probabilidad = binom.pmf(k, n, p)
print(f'{probabilidad:.8f}')

0.21284094


## Explicación por Mistral IA:
Este problema se trata de determinar la probabilidad de que un equipo de 12 miembros esté formado exactamente por 8 mujeres, dado que la proporción de mujeres participantes es del 60%. Para resolver esto, se utiliza la distribución binomial, que es adecuada para este tipo de problemas donde hay un número fijo de ensayos independientes (en este caso, la selección de miembros del equipo), cada uno con dos posibles resultados: éxito (seleccionar una mujer) o fracaso (seleccionar un hombre). 

In [24]:
p = 0.6
p

0.6

Aquí se define p como la probabilidad de éxito en cada ensayo, que en este contexto es la probabilidad de que un miembro del equipo sea mujer. El valor 0.6 representa el 60%.


In [25]:
n = 12
n

12

Aquí n es el número total de ensayos o miembros en un equipo. En este caso, cada equipo tiene 12 miembros.



In [26]:
k = 8
k

8

Aquí k es el número de éxitos deseados, es decir, el número de mujeres que queremos en el equipo. En este caso, queremos saber la probabilidad de tener exactamente 8 mujeres en un equipo.



In [27]:
from scipy.stats import binom

Esta línea importa la función binom del módulo scipy.stats, que se utiliza para trabajar con la distribución binomial.



In [28]:
probabilidad = binom.pmf(k, n, p)


Aquí se calcula la probabilidad de tener exactamente k éxitos (8 mujeres) en n ensayos (12 miembros) con una probabilidad de éxito p (60%). La función pmf (probability mass function) se utiliza para calcular la probabilidad de un valor exacto en una distribución binomial.



In [29]:
print(f'{probabilidad:.8f}')


0.21284094


Finalmente, esta línea imprime la probabilidad calculada con 8 decimales de precisión.

El código calcula la probabilidad de que un equipo de 12 miembros esté formado exactamente por 8 mujeres, dado que el 60% de los participantes son mujeres.

In [30]:
equipos = 30 * probabilidad
equipos

np.float64(6.3852281856)

La línea de código `equipos = 30 * probabilidad` se utiliza para estimar el número esperado de equipos que estarían formados exactamente por 8 mujeres, dado que hay un total de 30 equipos.

### Aquí está la explicación detallada:

* `probabilidad` es la probabilidad calculada de que un equipo de 12 miembros esté formado exactamente por 8 mujeres. Esta probabilidad se calculó utilizando la distribución binomial.

* Al multiplicar esta probabilidad por el número total de equipos (30), se obtiene una estimación del número esperado de equipos que cumplen con esta condición específica (tener exactamente 8 mujeres).

`En términos más simples`, si tienes una cierta probabilidad de que algo ocurra en un solo intento (en este caso, un equipo con 8 mujeres), y realizas múltiples intentos independientes (30 equipos), puedes estimar cuántas veces esperas que ese evento ocurra en total multiplicando la probabilidad por el número de intentos.

## 12 - Posibilidad de tener ojos azules

Suponga que la probabilidad de que una pareja tenga hijos con ojos azules es del 22%. En 50 familias, con 3 hijos cada una, ¿cuántas podemos esperar que tengan dos hijos con ojos azules?



## Solución de Alura + Mistral IA:

In [31]:
from scipy.stats import binom

# Probabilidad de tener un hijo con ojos azules
p = 0.22

# Número de hijos por familia
n = 3

# Número de hijos con ojos azules que queremos
k = 2

# Número total de familias
N = 50

# Calculamos la probabilidad de tener exactamente 2 hijos con ojos azules en una familia
probabilidad = binom.pmf(k, n, p)

# Número esperado de familias con exactamente 2 hijos con ojos azules
media = probabilidad * N

# Imprimimos los resultados
print(f'La probabilidad de que una familia tenga exactamente 2 hijos con ojos azules es: {probabilidad:.8f}')
print(f'Se espera que {media:.2f} familias tengan exactamente 2 hijos con ojos azules.')


La probabilidad de que una familia tenga exactamente 2 hijos con ojos azules es: 0.11325600
Se espera que 5.66 familias tengan exactamente 2 hijos con ojos azules.


### Por Mistral IA:
Para resolver este problema, seguiremos un enfoque similar al anterior. Utilizaremos la distribución binomial para calcular la probabilidad de que una familia con `3 hijos` tenga exactamente `2 hijos con ojos azules`, dado que la probabilidad de tener un hijo con ojos azules es del `22%`. Luego, multiplicaremos esta probabilidad por el número total de familias `(50)` para estimar cuántas familias se espera que tengan exactamente dos hijos con ojos azules.

Aquí está el código para resolver este problema:

In [32]:
# Probabilidad de tener un hijo con ojos azules
p = 0.22

# Número de hijos por familia
n = 3

# Número de hijos con ojos azules que queremos
k = 2

# Importamos la función de distribución binomial
from scipy.stats import binom

# Calculamos la probabilidad de tener exactamente 2 hijos con ojos azules en una familia
probabilidad = binom.pmf(k, n, p)

# Número esperado de familias con exactamente 2 hijos con ojos azules
familias_esperadas = 50 * probabilidad

print(f'La probabilidad de que una familia tenga exactamente 2 hijos con ojos azules es: {probabilidad:.8f}')
print(f'Se espera que {familias_esperadas:.2f} familias tengan exactamente 2 hijos con ojos azules.')


La probabilidad de que una familia tenga exactamente 2 hijos con ojos azules es: 0.11325600
Se espera que 5.66 familias tengan exactamente 2 hijos con ojos azules.


## 13 - Haga lo que hicimos en aula

Ha llegado el momento de que sigas todos los pasos que he dado durante esta clase. Si ya lo has hecho, excelente. Si no, es importante que realices lo visto en los videos para poder continuar con la siguiente clase.


## 14 - Lo que aprendimos

Lo que aprendimos en esta aula:

* **Los conceptos básicos de la distribución binomial de probabilidad**

* **Identificar y resolver problemas que involucran un experimento binomial**

* **Obtener la probabilidad de eventos que involucran sólo dos posibilidades de resultado**


---
---

## <font color=red>Problema</font>
***

Un restaurante recibe un promedio de **20 pedidos por hora**. ¿Cuál es la probabilidad de que, en un momento determinado al azar, el restaurante reciba **15 pedidos**?

## <font color=green>2.2 Distribución Poisson</font>
***

Se utiliza para describir el número de ocurrencias en un tiempo o espacio específico. Los eventos se caracterizan por la posibilidad de contar los éxitos, pero no poder contar los fracasos.

Como ejemplos de procesos donde podemos aplicar la distribución de Poisson tenemos la determinación del número de clientes que ingresan a una tienda en un momento determinado, el número de autos que llegan a un Auto-servicio de una cafetería a la hora del almuerzo, la determinación del número de accidentes registrados en un tramo de carretera, etc.

# $$P(k) = \frac{e^{-\mu}(\mu)^k}{k!}$$

Donde:

$e$ = constante cuyo valor aproximado es 2.718281828459045

$\mu$ = representa el número promedio de ocurrencias en un intervalo de tiempo o espacio dado

$k$ = número de éxitos en el intervalo deseado

### Experimento Poisson

1. La probabilidad de que ocurra es la misma en todo el intervalo observado.

2. El número de ocurrencias en un intervalo dado es independiente del número de ocurrencias en otros intervalos.

3. La probabilidad de que ocurra es la misma en intervalos de igual longitud.

### Media de la distribución Poisson

# $$\mu$$

### Desviación estándar de la distribución Poisson

# $$\sigma = \sqrt{\mu}$$

### Importando bibliotecas

http://www.numpy.org/

In [33]:
import numpy as np

In [34]:
np.e

2.718281828459045

## 03 - Experimento de Poisson

Las siguientes alternativas describen las características básicas de un experimento de Poisson.

Marque la que sea correcta.

`El número de ocurrencias en un intervalo dado es independiente del número de ocurrencias en otros intervalos`.

¡Alternativa correcta! La independencia entre experimentos es una característica importante en los cálculos de probabilidad.


## 04 - Calculando la probabilidad


## <font color='blue'>Ejemplo: Delivery</font>

Un restaurante recibe un promedio de **20 pedidos por hora**. ¿Cuál es la probabilidad de que, en un momento determinado al azar, el restaurante reciba **15 pedidos**?

### ¿Cuál es el número promedio de ocurrencias por hora ($\mu$)?

In [35]:
media = 20
media

20

### ¿Qué número de ocurrencias queremos obtener en el período ($k$)?

In [36]:
k = 15
k

15

### Solución 1

In [37]:
import numpy as np
import math

media = 20
k = 15

probabilidad = ((np.e ** (-media)) * (media ** k)) / (math.factorial(k))
print(f'{probabilidad}')


0.0516488535317584


### Importando bibliotecas

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.poisson.html

### Solución 2

In [38]:
from scipy.stats import poisson

In [39]:
probabilidad = poisson.pmf(k, media)
print(f'{probabilidad}')

0.05164885353175814


## 05 - Estimando los clientes en determinada hora

El número media de clientes que ingresan a una panadería por hora es 20. Obtenga la probabilidad de que, en la próxima hora, ingresen exactamente 25 clientes.


In [40]:
# Por Mistral IA:
from scipy.stats import poisson

# Número promedio de clientes por hora
media = 20

# Número de clientes para los que queremos calcular la probabilidad
k = 25

# Calculamos la probabilidad de que exactamente 25 clientes ingresen en la próxima hora
probabilidad = poisson.pmf(k, media)

print(f'La probabilidad de que exactamente 25 clientes ingresen en la próxima hora es: {probabilidad:.8f}')


La probabilidad de que exactamente 25 clientes ingresen en la próxima hora es: 0.04458765


## Explicación de Mistral IA:
El código que te proporcioné y el que mencionaste como correcto son esencialmente el mismo, solo que difieren en el formato de salida. Ambos utilizan la función `poisson.pmf` de la biblioteca `scipy.stats` para calcular la probabilidad de que exactamente `25` clientes ingresen a la panadería en una hora, dado un promedio de `20` clientes por hora.

### La diferencia está en cómo se imprime el resultado:

En el código que te proporcioné, la probabilidad se imprime como un número decimal con 8 lugares decimales: `0.04458765`.

En el código que mencionaste como correcto, la probabilidad se imprime como un porcentaje con 2 lugares decimales: `4.46%`.

Ambos resultados representan la misma probabilidad, solo que en formatos diferentes. La probabilidad de 0.04458765 es equivalente a 4.458765%, que se redondea a 4.46% cuando se imprime con dos decimales.

Por lo tanto, ambos códigos son correctos y proporcionan la misma información, solo que en formatos ligeramente diferentes.

---
---

## Por Alura:

In [41]:
from scipy.stats import poisson

media = 20
k = 25

probabilidad = poisson.pmf(k, media)
print(f'{probabilidad:.2%}')

4.46%


## 06 - Haga lo que hicimos en aula

Ha llegado el momento de que sigas todos los pasos que he dado durante esta clase. Si ya lo has hecho, excelente. Si no, es importante que realices lo visto en los videos para poder continuar con la siguiente clase.



## 07 - Lo que aprendimos

Lo que aprendimos en esta aula:

1. **Los conceptos básicos de la distribución de probabilidad Poisson**

2. **Obtener probabilidades en problemas como los siguientes:**

  * ***Determinar la cantidad de clientes que ingresan a una tienda en un momento dado***

  * ***Número de autos que llegan al parqueo de un restaurante a la hora del almuerzo***

  * ***Determinación del número de accidentes registrados en un tramo de carretera***

## <font color=red>Problema</font>
***

En un estudio sobre la altura de los residentes de una ciudad, se encontró que el conjunto de datos sigue una **distribución aproximadamente normal**, con **media 1.70** y **desviación estándar de 0.1**. Con esta información, obtenga el siguiente conjunto de probabilidades:

> **A.** probabilidad de que una persona, seleccionada al azar, sea menor de 1.80 metros.

> **B.** probabilidad de que una persona, seleccionada al azar, tenga entre 1.60 metros y 1.80 metros.

> **C.** probabilidad de que una persona, seleccionada al azar, tenga más de 1.90 metros.


## <font color=green>2.3 Distribución Normal</font>
***
## 02 - Conociendo la distribución Normal


La distribución normal es una de las más utilizadas en estadística. Es una distribución continua, donde la distribución de frecuencias de una variable cuantitativa tiene forma de campana y es simétrica en relación con su media.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img001.png' width='500px'>

### Características importantes

1. Es simétrica alrededor de la media;

2. El área bajo la curva corresponde a la proporción 1 o 100%;

3. Las medidas de tendencia central (media, mediana y moda) tienen el mismo valor;

4. Los extremos de la curva tienden al infinito en ambas direcciones y, teóricamente, nunca tocan el eje $x$;

5. La desviación estándar define la planitud y el ancho de la distribución. Las curvas más anchas y planas tienen valores de desviación estándar más altos;

6. La distribución se define por su media y desviación estándar;

7. La probabilidad siempre será igual al área bajo la curva, delimitada por los límites superior e inferior.

# $$f(x) = \frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Donde:

$x$ = variable normal

$\sigma$ = desviación estándar

$\mu$ = media

La probabilidad se obtiene del área bajo la curva, delimitada por los límites superior e inferior especificados. Se puede ver un ejemplo en la figura siguiente.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img002.png' width='500px'>


Para obtener el área anterior, simplemente calcule la integral de la función para los intervalos determinados. Según la siguiente ecuación:

# $$P(L_i<x<L_s) = \int_{L_i}^{L_s}\frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Donde:

$x$ = variable normal

$\sigma$ = desviación estándar

$\mu$ = media

$L_i$ = límite inferior

$L_s$ = límite superior

## 03 - Distribución de Probabilidad Normal



Con respecto a la distribución normal, califique las siguientes afirmaciones:

1) **La distribución normal se define por su media y desviación estándar.**
2) **La distribución normal es simétrica alrededor de la media.**
3) **La desviación estándar define el aplanamiento y el ancho de la curva de la distribución normal.**

**¿Cuáles son las correctas?**
 ### Todas las afirmaciones son correctas


¡Alternativa correcta! Perfecto, en nuestro notebook de clase, tenemos un conjunto de elementos que caracterizan la distribución normal.

## 04 - Usando la tabla estandarizada


Se crearon tablas estandarizadas para facilitar la obtención de los valores de las áreas bajo la curva normal y eliminar la necesidad de resolver integrales definidas.

Para ver los valores en una tabla estandarizada, simplemente transforme nuestra variable en una variable estandarizada $Z$.

Esta variable $Z$ representa la desviación en las desviaciones estándar de un valor de la variable original en relación con la media.

# $$Z = \frac{x-\mu}{\sigma}$$

Donde:

$x$ = variable normal con media $\mu$ y desviación estándar $\sigma$

$\sigma$ = desviación estándar

$\mu$ = media

### Construyendo la tabla normal estandarizada
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html

In [42]:
import pandas as pd
import numpy as np
from scipy.stats import norm

tabla_normal_estandarizada = pd.DataFrame(
    [], 
    index=["{0:0.2f}".format(i / 100) for i in range(0, 400, 10)],
    columns = ["{0:0.2f}".format(i / 100) for i in range(0, 10)])

for index in tabla_normal_estandarizada.index:
    for column in tabla_normal_estandarizada.columns:
        Z = np.round(float(index) + float(column), 2)
        tabla_normal_estandarizada.loc[index, column] = "{0:0.4f}".format(norm.cdf(Z))

tabla_normal_estandarizada.rename_axis('Z', axis = 'columns', inplace = True)

tabla_normal_estandarizada

Z       0.00    0.01    0.02    0.03    0.04    0.05    0.06    0.07    0.08  \
0.00  0.5000  0.5040  0.5080  0.5120  0.5160  0.5199  0.5239  0.5279  0.5319   
0.10  0.5398  0.5438  0.5478  0.5517  0.5557  0.5596  0.5636  0.5675  0.5714   
0.20  0.5793  0.5832  0.5871  0.5910  0.5948  0.5987  0.6026  0.6064  0.6103   
0.30  0.6179  0.6217  0.6255  0.6293  0.6331  0.6368  0.6406  0.6443  0.6480   
0.40  0.6554  0.6591  0.6628  0.6664  0.6700  0.6736  0.6772  0.6808  0.6844   
0.50  0.6915  0.6950  0.6985  0.7019  0.7054  0.7088  0.7123  0.7157  0.7190   
0.60  0.7257  0.7291  0.7324  0.7357  0.7389  0.7422  0.7454  0.7486  0.7517   
0.70  0.7580  0.7611  0.7642  0.7673  0.7704  0.7734  0.7764  0.7794  0.7823   
0.80  0.7881  0.7910  0.7939  0.7967  0.7995  0.8023  0.8051  0.8078  0.8106   
0.90  0.8159  0.8186  0.8212  0.8238  0.8264  0.8289  0.8315  0.8340  0.8365   
1.00  0.8413  0.8438  0.8461  0.8485  0.8508  0.8531  0.8554  0.8577  0.8599   
1.10  0.8643  0.8665  0.8686  0.8708  0.8729  0.8749  0.8770  0.8790  0.8810   
1.20  0.8849  0.8869  0.8888  0.8907  0.8925  0.8944  0.8962  0.8980  0.8997   
1.30  0.9032  0.9049  0.9066  0.9082  0.9099  0.9115  0.9131  0.9147  0.9162   
1.40  0.9192  0.9207  0.9222  0.9236  0.9251  0.9265  0.9279  0.9292  0.9306   
1.50  0.9332  0.9345  0.9357  0.9370  0.9382  0.9394  0.9406  0.9418  0.9429   
1.60  0.9452  0.9463  0.9474  0.9484  0.9495  0.9505  0.9515  0.9525  0.9535   
1.70  0.9554  0.9564  0.9573  0.9582  0.9591  0.9599  0.9608  0.9616  0.9625   
1.80  0.9641  0.9649  0.9656  0.9664  0.9671  0.9678  0.9686  0.9693  0.9699   
1.90  0.9713  0.9719  0.9726  0.9732  0.9738  0.9744  0.9750  0.9756  0.9761   
2.00  0.9772  0.9778  0.9783  0.9788  0.9793  0.9798  0.9803  0.9808  0.9812   
2.10  0.9821  0.9826  0.9830  0.9834  0.9838  0.9842  0.9846  0.9850  0.9854   
2.20  0.9861  0.9864  0.9868  0.9871  0.9875  0.9878  0.9881  0.9884  0.9887   
2.30  0.9893  0.9896  0.9898  0.9901  0.9904  0.9906  0.9909  0.9911  0.9913   
2.40  0.9918  0.9920  0.9922  0.9925  0.9927  0.9929  0.9931  0.9932  0.9934   
2.50  0.9938  0.9940  0.9941  0.9943  0.9945  0.9946  0.9948  0.9949  0.9951   
2.60  0.9953  0.9955  0.9956  0.9957  0.9959  0.9960  0.9961  0.9962  0.9963   
2.70  0.9965  0.9966  0.9967  0.9968  0.9969  0.9970  0.9971  0.9972  0.9973   
2.80  0.9974  0.9975  0.9976  0.9977  0.9977  0.9978  0.9979  0.9979  0.9980   
2.90  0.9981  0.9982  0.9982  0.9983  0.9984  0.9984  0.9985  0.9985  0.9986   
3.00  0.9987  0.9987  0.9987  0.9988  0.9988  0.9989  0.9989  0.9989  0.9990   
3.10  0.9990  0.9991  0.9991  0.9991  0.9992  0.9992  0.9992  0.9992  0.9993   
3.20  0.9993  0.9993  0.9994  0.9994  0.9994  0.9994  0.9994  0.9995  0.9995   
3.30  0.9995  0.9995  0.9995  0.9996  0.9996  0.9996  0.9996  0.9996  0.9996   
3.40  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997   
3.50  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998   
3.60  0.9998  0.9998  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.70  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.80  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.90  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000   

Z       0.09  
0.00  0.5359  
0.10  0.5753  
0.20  0.6141  
0.30  0.6517  
0.40  0.6879  
0.50  0.7224  
0.60  0.7549  
0.70  0.7852  
0.80  0.8133  
0.90  0.8389  
1.00  0.8621  
1.10  0.8830  
1.20  0.9015  
1.30  0.9177  
1.40  0.9319  
1.50  0.9441  
1.60  0.9545  
1.70  0.9633  
1.80  0.9706  
1.90  0.9767  
2.00  0.9817  
2.10  0.9857  
2.20  0.9890  
2.30  0.9916  
2.40  0.9936  
2.50  0.9952  
2.60  0.9964  
2.70  0.9974  
2.80  0.9981  
2.90  0.9986  
3.00  0.9990  
3.10  0.9993  
3.20  0.9995  
3.30  0.9997  
3.40  0.9998  
3.50  0.9998  
3.60  0.9999  
3.70  0.9999  
3.80  0.9999  
3.90  1.0000

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img003.png' width='250px'>

La tabla anterior proporciona el área bajo la curva entre $-\infty$ y $Z$ desviaciones estándar por encima de la media. Recordando que dado que estos son valores estandarizados, tenemos $\mu = 0$.

## 05 - Variable estandarizada Z

Hablando de tablas estandarizadas, vimos en nuestro último video que, para poder consultar este tipo de tablas, primero necesitamos realizar una transformación en nuestra variable original. La variable resultante de la transformación se conoce como `variable estandarizada Z`.

Marque la opción que muestra la forma correcta de realizar esta transformación. Suponga que X es nuestra variable original.

# $$Z = \frac{x-\mu}{\sigma}$$

Donde:

$x$ = variable normal con media $\mu$ y desviación estándar $\sigma$

$\sigma$ = desviación estándar

$\mu$ = media

donde `μ` es la media y `σ` es la desviación estándar de la variable `X`


## 06 - Utilizando la tabla - Caso 1


## <font color='blue'>Ejemplo: ¿Cuánto mides?</font>

En un estudio sobre la altura de los residentes de una ciudad, se encontró que el conjunto de datos sigue una **distribución aproximadamente normal**, con **media 1.70** y **desviación estándar de 0.1**. Con esta información, obtenga el siguiente conjunto de probabilidades:

> **A.** probabilidad de que una persona, seleccionada al azar, mida menos que 1.80 metros.

> **B.** probabilidad de que una persona, seleccionada al azar, mida entre 1.60 metros y 1.80 metros.

> **C.** probabilidad de que una persona, seleccionada al azar, mida más de 1.90 metros.

### Problema A - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img004.png' width='350px'>


### Obtener la variable estandarizada $Z$

In [43]:
media = 1.70
media

1.7

In [44]:
desviacion_estandar = 0.1
desviacion_estandar

0.1

In [45]:
z = (1.8 - media) / desviacion_estandar
z

1.0000000000000009

### Solución 1 - Utilizando la tabla

|Z|0.00|
|---|---|
|1.00|0.8413|

In [46]:
probabilidad = 0.8413
probabilidad

0.8413

### Solución 2 - Utilizando Scipy

In [47]:
from scipy.stats import norm
norm.cdf(z)

np.float64(0.8413447460685431)

## 07 - Prueba de Estadística

La aplicación de una prueba Estadística en una competencia presentó un conjunto de notas normalmente distribuidas. Se encontró que el `conjunto de notas tenía una media de 70` y una `desviación estándar de 5 puntos`.

¿Qué probabilidad hay de que un estudiante, seleccionado al azar, obtenga una puntuación inferior a `85 puntos`?


In [48]:
from scipy.stats import norm

media = 70
desviacion_estandar = 5
Z = (85 - media) / desviacion_estandar

norm.cdf(Z)

np.float64(0.9986501019683699)

## Mejora por Mistral IA:

In [49]:
from scipy.stats import norm

media = 70
desviacion_estandar = 5
Z = (85 - media) / desviacion_estandar

probabilidad = norm.cdf(Z)
print(f'{probabilidad:.4f}')  # Imprime la probabilidad en formato decimal
print(f'{probabilidad:.2%}')  # Imprime la probabilidad como porcentaje


0.9987
99.87%


## 08 - Utilizando la tabla - Caso 2


### Problema B - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img005.png' width='350px'>

### Obtener la variable estandarizada $Z$

In [50]:
import numpy as np

In [51]:
np.e

2.718281828459045

In [52]:
media = 1.70
z = (1.7 - media) / desviacion_estandar
z

0.0

In [53]:
desviacion_estandar = 0.1
desviacion_estandar

0.1

In [54]:
z = (1.8 - media) / desviacion_estandar
z

1.0000000000000009

### Solución 1 - Utilizando la tabla

|Z|0.00|
|---|---|
|1.00|0.8413|

In [55]:
probabilidad = 0.8413
probabilidad

0.8413

In [56]:
probabilidad = 0.8413 - 0.5 
probabilidad

0.34130000000000005

In [57]:
probabilidad = (0.8413 - 0.5) * 2 
probabilidad

0.6826000000000001

### Solución 2 - Utilizando Scipy

In [58]:
z_superior = (1.8 - media) / desviacion_estandar
z_superior


1.0000000000000009

In [59]:
z_inferior = (1.6 - media) / desviacion_estandar
z_inferior


-0.9999999999999987

In [60]:
probabilidad = norm.cdf(z_superior) - (1 - norm.cdf(z_superior))
print(f'{probabilidad:.4f}')  # Imprime la probabilidad en formato decimal

0.6827


In [61]:
probabilidad = norm.cdf(z_superior) - norm.cdf(z_inferior)
probabilidad

np.float64(0.6826894921370857)

## 09 - Facturación diaria

La facturación diaria de un controlador de aplicación sigue una distribución aproximadamente normal, con un media de $ 300,00 y una desviación estándar igual a $ 50,00. Obtenga las probabilidades de que, en un día aleatorio, el conductor gane:

* 1) Entre $ 250,00 y $ 350,00

* 2) Entre $ 400,00 y $ 500,00


## Solución 1:

In [62]:
from scipy.stats import norm

media = 300
desviacion_estandar = 50
Z_inferior = (250 - media) / desviacion_estandar
Z_superior = (350 - media) / desviacion_estandar

probabilidad = norm.cdf(Z_superior) - norm.cdf(Z_inferior)
print(f'{probabilidad:.2%}')

68.27%


## Solución 2:

In [63]:
from scipy.stats import norm

media = 300
desviacion_estandar = 50
Z_inferior = (400 - media) / desviacion_estandar
Z_superior = (500 - media) / desviacion_estandar

probabilidad = norm.cdf(Z_superior) - norm.cdf(Z_inferior)
print(f'{probabilidad:.2%}')

2.27%


## 10 - Utilizando la tabla - Caso 3


### Problema C - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img006.png' width='350px'>

### Obtener la variable estandarizada $Z$

In [67]:
media = 1.7
media

1.7

In [68]:
desviacion_estandar = 0.1
desviacion_estandar

0.1

In [69]:
z = (1.9 - media) / desviacion_estandar 
z

1.9999999999999996

### Solución 1 - Utilizando la tabla

|z|0.00|0.01|0.02|0.03|0.04|0.05|0.06|0.07|0.08|0.09|
|:---:|---|---|---|---|---|---|---|---|---|---|
|1.90||||||||||0.9767|
|2.00|0.977||||||||||

![z.png](/home/pol/Escritorio/ipynb-alura/assets/z.png)

In [ ]:
probabilidad = 1 - 0.9767 # z=1.90-0.09
probabilidad

0.023299999999999987

In [ ]:
probabilidad = 1 - 0.9777 # z=2.00-0.00 -> "Más cerca de Scipy"
probabilidad

0.022299999999999986

### Solución 2 - Utilizando Scipy

In [ ]:
probabilidad = 1 - norm.cdf(z) # -1 porque es la zona de la derecha
probabilidad

np.float64(0.02275013194817921)

In [73]:
probabilidad = norm.cdf(-z) 
probabilidad

np.float64(0.022750131948179216)

## 11 - Vida útil

Una empresa verificó que las lámparas incandescentes XPTO tienen una vida útil normalmente distribuida, con un `media igual a 720 días` y una `desviación estándar igual a 30 días`. Calcula la `probabilidad` de que una lámpara, elegida al azar, dure:

* 1) Entre `650` y `750` días

* 2) Más de `800` días

+ 3) Menos de `700` días



In [76]:
from scipy.stats import norm

media = 720
desviacion_estandar = 30

# Item A
Z_inferior = (650 - media) / desviacion_estandar
Z_superior = (750 - media) / desviacion_estandar

probabilidad = norm.cdf(Z_superior) - norm.cdf(Z_inferior)
print(f'{probabilidad:.2%}')

# Item B
Z = (800 - media) / desviacion_estandar

probabilidad = 1 - norm.cdf(Z)
print(f'{probabilidad:.2%}')

# Item C
Z = (700 - media) / desviacion_estandar

probabilidad = norm.cdf(Z)
print(f'{probabilidad:.2%}')

83.15%
0.38%
25.25%


## 12 - Calculando probabilidades

Usando la tabla estandarizada, o las herramientas proporcionadas por Python, encuentre el área debajo de la curva normal para los valores Z a continuación:

1\)- Z < 1,96

2\)- Z > 2,15

3\)- Z < -0,78

4\)- Z > 0,59


In [ ]:
from scipy.stats import norm

# Item A
probabilidad = norm.cdf(1.96)
print(f'{probabilidad:0.4f}')

# Item B
probabilidad = 1 - norm.cdf(2.15)
# o -> probabilidad = norm.sf(2.15)
print(f'{probabilidad:0.4f}')

# Item C
probabilidad = norm.cdf(-0.78)
print(f'{probabilidad:0.4f}')

# Item D
probabilidad = 1 - norm.cdf(0.59)
# o -> probabilidad = norm.sf(0.59)
print(f'{probabilidad:0.4f}')


0.9750
0.0158
0.2177
0.2776


Puse un comentario en el código para avanzar en un tema que veremos en el próximo entrenamiento de estadística. El método `sf()` de norm sirve para obtener el complemento de la probabilidad reportada por el método `cdf()`, es decir, `norm.sf(Z) = 1 - norm.cdf(Z)`.

## 13 - Haga lo que hicimos en aula

Ha llegado el momento de que sigas todos los pasos que he dado durante esta clase. Si ya lo has hecho, excelente. Si no, es importante que realices lo visto en los videos para poder continuar con la siguiente clase.


## 14 - Lo que aprendimos

Lo que aprendimos en esta aula:

* **Los conceptos básicos de la distribución de probabilidad Normal** 

* **Trabajar con la tabla estandarizada Z**

* **Obtener las probabilidades usando la distribución normal en un conjunto de situaciones**

# <font color=green>04 - MUESTREO</font>
***

## <font color=green>3.1 Población y Muestra</font>
***

### Población
<h2><u>Población</u></h2>
Conjunto de todos los elementos de interés de un estudio. Varios elementos pueden componer una población, por ejemplo: personas, edades, alturas, automóviles, etc.

En cuanto al tamaño, las poblaciones pueden ser limitadas (poblaciones finitas) o ilimitadas (poblaciones infinitas).

### Poblaciones finitas

Permiten el recuento de sus elementos. Los ejemplos incluyen el número de empleados en una empresa, el número de estudiantes en una escuela, etc.

### Poblaciones infinitas

No es posible contar sus elementos. Como ejemplos tenemos el número de porciones que se pueden extraer del agua del mar para su análisis, la temperatura medida en cada punto de un territorio, etc.

<font color=red>Cuando se pueden contar los elementos de una población, pero presentando una cantidad muy grande, se supone que la población es infinita.</font>

### Muestra
Subconjunto representativo de la población.

Los atributos numéricos de una población, como su media, varianza y desviación estándar, se conocen como **parámetros**. El enfoque principal de la inferencia estadística es precisamente generar estimaciones y probar hipótesis sobre parámetros poblacionales utilizando información de la muestra.

## <font color=green>3.2 ¿Cuándo usar una muestra?</font>
***

### Poblaciones infinitas

El estudio nunca llegaría a su fin. No es posible investigar todos los elementos de la población.

### Pruebas destructivas

Estudios donde los elementos evaluados son totalmente consumidos o destruidos. Ejemplo: pruebas de vida útil, pruebas de seguridad en accidentes de tráfico.

### Resultados rápidos

Investigaciones que necesitan más agilidad en la difusión. Ejemplo: encuestas de opinión, encuestas sobre problemas de salud pública.

### Altos precios

Cuando la población es finita pero muy grande, el costo de un censo puede hacer inviable el proceso.

## <font color=green>3.3 Muestreo Aleatorio Simple</font>
***

Es una de las principales formas de extraer una muestra de una población. El requisito fundamental de este tipo de enfoque es que cada elemento de la población tenga las mismas posibilidades de ser seleccionado para formar parte de la muestra.

## <font color=green>3.4 Muestreo Estratificado</font>
***

Es una mejora del proceso de muestreo aleatorio simple. En este método, se propone dividir la población en subgrupos de elementos con características similares, es decir, grupos más homogéneos. Con estos subgrupos separados, la técnica de muestreo aleatorio simple dentro de cada subgrupo se aplica individualmente.

## <font color=green>3.5 Muestreo por Conglomerados</font>
***

También tiene como objetivo mejorar el criterio de muestreo aleatorio simple. En el muestreo por conglomerados también se crean subgrupos, pero no serán homogéneos como en el muestreo estratificado. En el muestreo por conglomerados los subgrupos serán heterogéneos, donde luego se aplicará un muestreo aleatorio simple o estratificado.

Un ejemplo muy común de la aplicación de este tipo de técnica es la división de la población en grupos territoriales, donde los elementos investigados tendrán características bastante variadas.

# <font color=green>4 ESTIMACIÓN</font>
***

## <font color='red'>Problema </font>

Suponga que los pesos de las bolsas de arroz de una industria alimentaria se distribuyen aproximadamente como una distribución normal **desviación estándar igual a 150 g**. Se seleccionó una **muestra aleatoria de 20 bolsas** de un lote específico, se obtuvo un **peso promedio de 5.050 g**. Construya un **intervalo de confianza para la media de la población** asumiendo un **nivel de significancia del 5%**.

---

Es la forma de hacer suposiciones generalizadas sobre los parámetros de una población a partir de información de una muestra.

- **Los parámetros** son los atributos numéricos de una población, como la media, la desviación estándar, etc.

- **Estimación** es el valor obtenido para un parámetro dado a partir de los datos de una muestra de la población.

## <font color=green>4.1 Teorema central del límite</font>
***

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

La desviación estándar de las medias muestrales se conoce como **error estándar de la media**

### Entendiendo el Teorema Central del Límite

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

## <font color=green>4.2 Los niveles de confianza y significación</font>
***

El **nivel de confianza** ($1 - \alpha$) representa la probabilidad de que la estimación sea correcta. Además, el **nivel de significancia** ($\alpha$) expresa la probabilidad de error de la estimación.

El **nivel de confianza** representa el grado de confiabilidad del resultado estimado dentro de un cierto rango. Cuando establecemos un **nivel de confianza del 95%** en una encuesta, por ejemplo, estamos asumiendo que existe una probabilidad del 95% de que los resultados de la encuesta representen bien la realidad, es decir, sean correctos.

El **nivel de confianza** de una estimación se puede obtener del área bajo la curva normal como se muestra en la figura siguiente.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img007.png' width='450px'>


## <font color=green>4.3 Error inferencial</font>
***

El **error inferencial** se define por la **desviación estándar de las medias muestrales** $\sigma_\bar{x}$ y por el **nivel de confianza** determinado para el proceso.

# $$e = z \frac{\sigma}{\sqrt{n}}$$

## <font color=green>4.4 Intervalos de confianza</font>
***

### Intervalo de confianza para la media poblacional

#### Con desviación estándar poblacional conocida

## $$\mu = \bar{x} \pm z\frac{\sigma}{\sqrt{n}}$$

#### Con desviación estándar de población desconocida

## $$\mu = \bar{x} \pm z\frac{s}{\sqrt{n}}$$

## <font color='blue'>Ejemplo: </font>

Suponga que los pesos de las bolsas de arroz de una industria alimentaria se distribuyen aproximadamente como una población normal con **desviación estándar igual a 150 g**. Se seleccionó una **muestra aleatoria de 20 bolsas** de un lote específico, se obtuvo un **peso promedio de 5.050 g**. Construya un intervalo de confianza para el **promedio de la población** asumiendo un **nivel de significancia del 5%**.

### Media muestral

### Nivel de significancia ($\alpha$)

### Nivel de confianza ($1 - \alpha$)

### Obteniendo $z$

### Obteniendo $z$

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img008.png' width='450px'>

### Valores de $z$ para los niveles de confianza más utilizados

|Nivel de<br>confianza|Valor del área bajo<br>la curva normal| $z$ |
|:----------------:|:---------------------------------:|:---:|
|90%               |0.95                               |1.645|
|95%               |0.975                              |1.96 |
|99%               |0.995                              |2.575|

### Obteniendo $\sigma_\bar{x}$

### Obteniendo $e$

### Solución 1 - Calculando el intervalo de confianza para la media

### Solución 2 - Calculando el intervalo de confianza para la media

# <font color=green>5 CÁLCULO DEL TAMAÑO DE LA MUESTRA</font>
***

## <font color='red'>Problema </font>

Estamos estudiando los ingresos mensuales de los jefes de hogar con ingresos de hasta R$\$$ 5.000,00 en Brasil. Nuestro supervisor determinó que el **error máximo en relación a la media es de R$\$$ 10,00**. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **R$\$$ 1.082,79**. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio?

## <font color=green>5.1 Variables cuantitativas y población infinita</font>
***

# $$e = z \frac{\sigma}{\sqrt{n}}$$

#### Con desviación estándar conocida

## $$n = \left(z\frac{\sigma}{e}\right)^2$$

#### Con desviación estándar desconocida

## $$n = \left(z\frac{s}{e}\right)^2$$

Donde:

$z$ = variable normal estandarizada

$\sigma$ = desviación estándar poblacional

$s$ = desviación estándar de la muestra

$e$ = error inferencial

### <font color='red'>Comentarios</font>

1. La desviación estándar ($\sigma$ o $s$) y el error ($e$) deben estar en la misma unidad de medida.

2. Cuando el error ($e$) se representa en términos porcentuales, debe interpretarse como un porcentaje relacionado con el promedio.

## <font color='blue'>Ejemplo: Rendimiento medio</font>

Estamos estudiando los ingresos mensuales de los jefes de hogar. Nuestro supervisor determinó que el **error máximo en relación a la media es de $\$$ 100,00** pesos. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **$\$$ 3.323,39** pesos. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio?

### Obteniendo $\sigma$

### Obteniendo $e$

### Obteniendo $n$

---
---

## <font color='red'>Problema</font>

En un lote de **10 000 latas** de gaseosa, se realizó una muestra aleatoria simple de **100 latas** y se obtuvo que la **desviación estándar muestral del contenido de las latas es igual a 12 ml**. El fabricante estipula un **error máximo en la media poblacional de solo 5 ml**. Para garantizar un **nivel de confianza del 95%**, ¿qué tamaño de muestra se debe seleccionar para este estudio?

## <font color=green>5.2 Variables cuantitativas y población finita</font>
***

#### Con desviación estándar conocida

## $$n = \frac{z^2 \sigma^2 N}{z^2 \sigma^2 + e^2(N-1)}$$

#### Con desviación estándar desconocida

## $$n = \frac{z^2 s^2 N}{z^2 s^2 + e^2(N-1)}$$

Donde:

$N$ = tamaño de la población

$z$ = variable normal estandarizada

$\sigma$ = desviación estándar poblacional

$s$ = desviación estándar muestral

$e$ = error inferencial

## <font color='blue'>Ejemplo: Industria de refrescos</font>

En un lote de **10 000 latas** de gaseosa, se realizó una muestra aleatoria simple de **100 latas** y se obtuvo que la **desviación estándar muestral del contenido de las latas es igual a 12 ml**. El fabricante estipula un **error máximo en la media poblacional de solo 5 ml**. Para garantizar un **nivel de confianza del 95%**, ¿qué tamaño de muestra se debe seleccionar para este estudio?

### Obteniendo $N$

### Obteniendo $z$

### Obteniendo $s$

### Obteniendo $e$

### Obteniendo $n$

## $$n = \frac{z^2 s^2 N}{z^2 s^2 + e^2(N-1)}$$

# <font color=green>6 FIJANDO EL CONTENIDO</font>
***

## <font color='blue'>Ejemplo: Rendimiento medio</font>

Estamos estudiando **los ingresos mensuales de los jefes de hogar con ingresos de hasta R$\$$ 5.000,00 en Brasil**. Nuestro supervisor determinó que el **error máximo en relación a la media es de R$\$$ 10,00**. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **R$\$$ 1.082,79** y que la **media poblacional** es **R$\$$ 1.426,54**. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio? ¿Cuál es el intervalo de confianza para la media considerando el tamaño de muestra obtenido?

### Construyendo el conjunto de datos según lo especificado por el problema

### Calcular el tamaño de la muestra

### Calcular el intervalo de confianza para la media

### Realización de una prueba gráfica